src/render/md.jl

```diff
@@ -43,50 +56,84 @@ function writemd(io::IO, doc::Metadata, config::Config)
+                        anchornum = basenames[basename] += 1 :
+                        anchornum = basenames[basename] = 1
+                push!(ents, modulename(doc), obj, entries(doc)[obj],
+                        "$(string(k))__$(basename).$(anchornum)", k)
+            end
+        end
+    else
+        for k in config.category_order
+            haskey(index, k) || continue
+            for (s, obj) in index[k]
+                push!(ents, modulename(doc), obj, entries(doc)[obj], "", k)
             end
         end
-        println(io)
-        writemd(io, ents, config)
     end
+    return ents
 end
 
-function writemd(io::IO, ents::Entries, config::Config)
-    exported = Entries()
-    internal = Entries()
-
-    for (modname, obj, ent) in ents.entries
-        isexported(modname, obj) ?
-            push!(exported, modname, obj, ent) :
-            config.include_internal && push!(internal, modname, obj, ent)
-    end
-
-    if !isempty(exported.entries)
-        print_help(io, config.mdstyle_exported, "Exported")
-        for (modname, obj, ent) in exported.entries
-            writemd(io, modname, obj, ent, config)
-        end
-    end
-    if !isempty(internal.entries)
-        print_help(io, config.mdstyle_internal, "Internal")
-        for (modname, obj, ent) in internal.entries
-            writemd(io, modname, obj, ent, config)
+function process_entriesmd(io::IO, grpname::ByteString,
+                           entries::Dict{Symbol, Vector{(Module, Any, Entry, ByteString)}},
+                           config::Config)
+    if has_items(entries)
+        config.md_subheader == :SIMPLE && println_mdstyle(io, config.mdstyle_subheader, grpname)
+        for k in config.category_order
+            if length(entries[k]) > 0
+                config.md_subheader == :CATEGORY && println_mdstyle(io, config.mdstyle_subheader,
+                                                              "$(ucfirst(string(k)))s [$grpname]\n")
+                for (modname, obj, ent, anchorname) in entries[k]
+                    writemd(io, modname, obj, ent, anchorname, config)
+                end
+            end
         end
     end
 end
 
-function writemd{category}(io::IO, modname, obj, ent::Entry{category}, config::Config)
+function writemd{category}(io::IO, modname, obj, ent::Entry{category}, anchorname::String,
+                                                                            config::Config)
     objname = writeobj(obj, ent)
     println(io, "---\n")
-    print_help(io, config.mdstyle_objname, objname)
+    (config.md_genindex || config.md_permalink) &&
+            println(io, """<a id="$anchorname" class="lexicon_definition"></a>""")
+
+    config.md_permalink                                                                 ?
+            println_mdstyle(io, config.mdstyle_objname, "$objname   [¶](#$anchorname)") :
+            println_mdstyle(io, config.mdstyle_objname, objname)
```

Comment:
Since only the last arguments are affected by the condition `config.md_permalink` this could be reduced to:

```julia
println_mdstyle(io, config.mdstyle_objname,
    string(objname, config.md_permalink ? " [¶](#$anchorname)") : "")
```